pkt-meta/src/flow_table/table.rs

```diff
@@ -0,0 +1,678 @@
+    use net::vxlan::Vni;
+
+    use crate::flow_table::{FlowKey, FlowKeyData, IpProtoKey, TcpProtoKey};
+
+    #[concurrency_mode(std)]
+    mod std_tests {
+        use super::*;
+
+        #[test]
+        fn test_flow_table_insert_and_remove() {
+            let now = Instant::now();
+            let five_seconds = Duration::new(5, 0);
+            let five_seconds_from_now = now + five_seconds;
+
+            let flow_table = FlowTable::default();
+            let flow_key = FlowKey::Unidirectional(FlowKeyData::new(
+                Some(VpcDiscriminant::VNI(Vni::new_checked(1).unwrap())),
+                "1.2.3.4".parse::<IpAddr>().unwrap(),
+                Some(VpcDiscriminant::VNI(Vni::new_checked(2).unwrap())),
+                "4.5.6.7".parse::<IpAddr>().unwrap(),
+                IpProtoKey::Tcp(TcpProtoKey {
+                    src_port: TcpPort::new_checked(1025).unwrap(),
+                    dst_port: TcpPort::new_checked(2048).unwrap(),
+                }),
+            ));
+
+            let flow_info = FlowInfo::new(five_seconds_from_now);
+
+            flow_info.locked.write().unwrap().dst_vpcd =
+                Some(VpcDiscriminant::VNI(Vni::new_checked(2).unwrap()));
+
+            flow_table.insert(flow_key, flow_info);
+            let result = flow_table.remove(&flow_key).unwrap();
+            assert!(result.0 == flow_key);
+            assert_eq!(
+                result.1.locked.read().unwrap().dst_vpcd,
+                Some(VpcDiscriminant::VNI(Vni::new_checked(2).unwrap()))
+            );
+        }
+
+        #[test]
+        fn test_flow_table_timeout() {
+            let now = Instant::now();
+            let two_seconds = Duration::from_secs(2);
+            let one_second = Duration::from_secs(1);
+
+            let flow_table = FlowTable::default();
+            let flow_key = FlowKey::Unidirectional(FlowKeyData::new(
+                Some(VpcDiscriminant::VNI(Vni::new_checked(42).unwrap())),
+                "10.0.0.1".parse::<IpAddr>().unwrap(),
+                Some(VpcDiscriminant::VNI(Vni::new_checked(43).unwrap())),
+                "10.0.0.2".parse::<IpAddr>().unwrap(),
+                IpProtoKey::Tcp(TcpProtoKey {
+                    src_port: TcpPort::new_checked(1234).unwrap(),
+                    dst_port: TcpPort::new_checked(5678).unwrap(),
+                }),
+            ));
+
+            let flow_info = FlowInfo::new(now + two_seconds);
+            flow_table.insert(flow_key, flow_info);
+
+            // Wait 1 second, should still be present
+            thread::sleep(one_second);
+            // Reap expired entries after 1 second (should not reap our entry)
+            flow_table.reap_expired();
+            assert!(
+                flow_table.lookup(&flow_key).is_some(),
+                "Flow key should still be present after 1 second"
+            );
+
+            // Wait another 2 seconds (total 3s), should be expired
+            thread::sleep(two_seconds);
```

Comment:
Do we need full seconds (and adding 3 seconds each time we run the tests locally)? Or could we make these durations shorter?

Reply:
The test was a bit sensitive to local load on my workstation (compiles mainly) subsecond so I went to whole seconds.  I don't know

Reply:
I'm going to leave this for now until we decide how to deal with timeout test.  We really need simulated time.